src/js/controller/captions.js

```diff
@@ -108,7 +109,8 @@ const Captions = function(_model) {
         return list;
     }
 
-    function _selectDefaultIndex() {
+    function _selectDefaultIndex(captionsByDefault) {
+        _captionsByDefault = _captionsByDefault || captionsByDefault;
```

Comment:
Instead of the `||` just always call `_selectDefaultIndex(_captionsByDefault)` in this module.

Only use `captionsByDefault` in this function.
`_captionsByDefault = captionsByDefault;`